src/main/java/telraam/monitoring/BatonStatusHolder.java

```diff
@@ -19,14 +19,16 @@ public class BatonStatusHolder {
 
     private BatonDAO batonDAO;
     private DetectionDAO detectionDAO;
+    private boolean initialized = false;
 
     public BatonStatusHolder(BatonDAO BDAO, DetectionDAO DDAO) {
         batonDAO = BDAO;
         detectionDAO = DDAO;
-        this.initStatus();
     }
 
     private void initStatus() {
+        if (initialized) return;
```

Comment:
Since this is only executed once, if new batons are added at runtime (either via the DB or via the API), I think the code will crash then. I think this would be more cleanly solved with a `defaultdict`. Unfortunately, Java does not have this, but https://stackoverflow.com/questions/1786206/is-there-a-java-equivalent-of-pythons-defaultdict